### PR TITLE
Scala app structure

### PR DESCRIPTION
**Pull request by janiskomuls**

### CODE REVIEW

File: structure/build.sbt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+scalaVersion := "2.13.6"
+
+version := "1.0"
+
+libraryDependencies ++= Dependencies.All
+
+lazy val root = project
+  .enablePlugins(JUnitXmlReportPlugin)
```
Comment (mr-git) [inline structure/build.sbt:8]: this plugin is enabled by default in SBT

File: structure/docs/5-pkg-by-feature-pros-cons.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Package by feature/component
+
+- Pros
+    - Fixes package by type/layer Cons
+- Cons
```
Comment (mr-git) [inline structure/docs/5-pkg-by-feature-pros-cons.md:5]: here `Cons` is second, but in previous `Cons` was first item

File: structure/project/plugins.sbt [hunk 3]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.2")
```
Comment (mr-git) [inline structure/project/plugins.sbt:1]: isn't 2.4.3 latest?

File: structure/src/main/scala/com/evolution/controllers/UserController.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package com.evolution.controllers
+
+import com.evolution.domain._
```
Comment (mr-git) [inline structure/src/main/scala/com/evolution/controllers/UserController.scala:3]: shouldn't we set:
```scala 
      scalacOptions := Seq("-Xsource:3")
```
and amend `.scalafmt.conf` with:
```
runner.dialect = Scala213Source3
```
and here use future-proof `import com.evolution.domain.*`?

File: structure/docs/5-pkg-by-feature-pros-cons.md (hunk 2 above)
Comment (janiskomuls) [reply structure/docs/5-pkg-by-feature-pros-cons.md:5]: in lecture is will be from top to bottom, work with layered app, see Cons, mention some Pros, then refactor to feature and see Pros, then mention Cons

File: structure/build.sbt (hunk 1 above)
Comment (janiskomuls) [reply structure/build.sbt:8]: this was just to have example in sbt, but I better to remove it as this is not related to lecture much

**Pull request merged**